lib/elixir/lib/access.ex

```diff
@@ -829,4 +829,114 @@ defmodule Access do
   end
+
+  @doc ~S"""
+  Returns a function that accesses all items of a list that are within the provided range.
+
+  The range will be normalized following the same rules from `Enum.slice/2`.
+
+  The returned function is typically passed as an accessor to `Kernel.get_in/2`,
+  `Kernel.get_and_update_in/3`, and friends.
+
+  ## Examples
+
+      iex> list = [%{name: "john", salary: 10}, %{name: "francine", salary: 30}, %{name: "vitor", salary: 25}]
+      iex> get_in(list, [Access.slice(1..2), :name])
+      ["francine", "vitor"]
+      iex> get_and_update_in(list, [Access.slice(1..3//2), :name], fn prev ->
+      ...>   {prev, String.upcase(prev)}
+      ...> end)
+      {["francine"], [%{name: "john", salary: 10}, %{name: "FRANCINE", salary: 30}, %{name: "vitor", salary: 25}]}
+
+  `slice/1` can also be used to pop elements out of a list or
+  a key inside of a list:
+
+      iex> list = [%{name: "john", salary: 10}, %{name: "francine", salary: 30}, %{name: "vitor", salary: 25}]
+      iex> pop_in(list, [Access.slice(-2..-1)])
+      {[%{name: "francine", salary: 30}, %{name: "vitor", salary: 25}], [%{name: "john", salary: 10}]}
+      iex> pop_in(list, [Access.slice(-2..-1), :name])
+      {["francine", "vitor"], [%{name: "john", salary: 10}, %{salary: 30}, %{salary: 25}]}
+
+  When no match is found, an empty list is returned and the update function is never called
+
+      iex> list = [%{name: "john", salary: 10}, %{name: "francine", salary: 30}, %{name: "vitor", salary: 25}]
+      iex> get_in(list, [Access.slice(5..10//2), :name])
+      []
+      iex> get_and_update_in(list, [Access.slice(5..10//2), :name], fn prev ->
+      ...>   {prev, String.upcase(prev)}
+      ...> end)
+      {[], [%{name: "john", salary: 10}, %{name: "francine", salary: 30}, %{name: "vitor", salary: 25}]}
+
+  An error is raised if the accessed structure is not a list:
+
+      iex> get_in(%{}, [Access.slice(2..10//3)])
+      ** (ArgumentError) Access.slice/1 expected a list, got: %{}
+
+  An error is raised if the step of the range is negative:
+
+      iex> get_in([], [Access.slice(2..10//-1)])
+      ** (ArgumentError) Access.slice/1 does not accept ranges with negative steps, got: 2..10//-1
+
+  """
+  @doc since: "1.14"
+  @spec slice(Range.t()) :: access_fun(data :: list, current_value :: list)
+  def slice(%Range{} = range) do
+    if range.step > 0 do
+      fn op, data, next -> slice(op, data, range, next) end
+    else
+      raise ArgumentError,
+            "Access.slice/1 does not accept ranges with negative steps, got: #{inspect(range)}"
+    end
+  end
+
+  defp slice(:get, data, %Range{} = range, next) when is_list(data) do
+    data
+    |> Enum.slice(range)
+    |> Enum.map(next)
+  end
+
+  defp slice(:get_and_update, data, range, next) when is_list(data) do
+    range = normalize_range(range, data)
+    index = if(range.first >= 0, do: 0, else: length(data) * -1)
+    get_and_update_slice(data, range, next, [], [], index)
```

Comment:
```suggestion
    %{first: first, last: last} = normalize_range(range, data)

    if last < first do
      {[], data}
    else
      get_and_update_slice(data, range, next, [], [], index)
    end
```

You can add a test by checking that `Access.slice(2..1)` doesn't really access anything. :)

Reply:
Great! That could be done in normalize too? So we don't ever run `length(list)`

Reply:
nvm...

Reply:
Done!